client/models/model_editor.js

```diff
@@ -138,27 +138,55 @@ class Models {
         this.selected_type = 'model';
         this.selected_model = model;
 
+        this.textureSelect();
         this.resetUI();
-    }
 
+    }
+    
     loadAsset(id) {
         if (!id) return;
         if (this.selected_model == id) return;
-
+        
         let pack = this.loaded_asset_pack;
-
+        
         this.selected_type = 'asset';
         this.selected_model = id;
         this.asset_definition = {
             model: 'OBJ/' + id,
             texture: Object.keys(this.asset_packs[pack].textures)[0]
         }
-
+        
         this.needs_resize = true;
-
+        
+        this.textureSelect();
         this.resetUI();
     }
 
+    textureSelect(){
+        let select = document.getElementById("model-dialog-controls-texture");
+        select.options.length = 0;
+
+        let pack = this.loaded_asset_pack;
+
+        this.selected_type === 'asset'
+            ? createList(select, this.asset_packs[pack].textures)
```

Comment:
Pass 'false' as the third argument to these lists -- 'delete' is not a valid texture name.

Reply:
I passed True because the default for skip_delete is false

Reply:
True is right. The original PR wasn't passing anything.